app/src/sharedTest/java/org/oppia/android/app/parser/ListItemLeadingMarginSpanTest.kt

```diff
@@ -446,6 +475,235 @@ class ListItemLeadingMarginSpanTest {
+    val dir = 15
+    val top = 0
+    val bottom = 0
+    val htmlParser = htmlParserFactory.create(
+      resourceBucketName,
+      entityType = "",
+      entityId = "",
+      imageCenterAlign = true,
+      displayLocale = displayLocale
+    )
+    val htmlResult = activityScenarioRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_list_content_text_view)
+      return@runWithActivity htmlParser.parseOppiaHtml(
+        "<p>" +
+          "<ol>" +
+          "        <li> Usage Data\", such as:" +
+          "          <ol>" +
+          "            <li>your answers to Lessons;</li>" +
+          "            <li>when you begin and end a Lesson;</li>" +
+          "          </ol>" +
+          "        </li>" +
+          "        <li> any contributions you make to the Site (such as feedback on" +
+          "            Lessons, edits to Lessons, and Lessons created);</li>" +
+          "</ol>" +
+          "   </p>",
+        textView
+      )
+    }
+
+    /* Reference: https://medium.com/androiddevelopers/spantastic-text-styling-with-spans-17b0c16b4568#e345 */
+    val bulletSpans =
+      htmlResult.getSpans(
+        0,
+        htmlResult.length,
+        ListItemLeadingMarginSpan.OlSpan::class.java
+      )
+    assertThat(bulletSpans.size.toLong()).isEqualTo(4)
+
+    val bulletSpan0 = bulletSpans[0] as ListItemLeadingMarginSpan.OlSpan
+    assertThat(bulletSpan0).isNotNull()
+    htmlResult.setSpan(
+      bulletSpan0,
+      0,
+      3,
+      Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
+    )
+    bulletSpan0.drawLeadingMargin(
+      canvas, paint, x, dir, top, 78, bottom, htmlResult,
+      htmlResult.getSpanStart(bulletSpan0),
+      htmlResult.getSpanEnd(bulletSpan0),
+      true,
+      null
+    )
+    val bulletSpan1 = bulletSpans[1] as ListItemLeadingMarginSpan.OlSpan
+    bulletSpan1.drawLeadingMargin(
+      canvas, paint, x, dir, top, 164, bottom, htmlResult,
+      htmlResult.getSpanStart(bulletSpan1),
+      htmlResult.getSpanEnd(bulletSpan1),
+      true,
+      null
+    )
+
+    val bulletSpan2 = bulletSpans[2] as ListItemLeadingMarginSpan.OlSpan
+    assertThat(bulletSpan2).isNotNull()
+    bulletSpan2.drawLeadingMargin(
+      canvas, paint, x, dir, top, 250, bottom, htmlResult,
+      htmlResult.getSpanStart(bulletSpan2),
+      htmlResult.getSpanEnd(bulletSpan2),
+      true,
+      null
+    )
+    val bulletSpan3 = bulletSpans[3] as ListItemLeadingMarginSpan.OlSpan
+    assertThat(bulletSpan3).isNotNull()
+    bulletSpan3.drawLeadingMargin(
+      canvas, paint, x, dir, top, 422, bottom, htmlResult,
+      htmlResult.getSpanStart(bulletSpan3),
+      htmlResult.getSpanEnd(bulletSpan3),
+      true,
+      null
+    )
+
+    val shadowCanvas = shadowOf(canvas)
+
+    assertThat(shadowCanvas.textHistoryCount).isEqualTo(4)
+    // The below assertion verifies the 1st item of inner level list
+    assertThat(shadowCanvas.getDrawnTextEvent(0).x).isWithin(1e-5f).of(-96.0f)
```

Comment:
Ditto for these x coords--they don't quite seem correct.

Reply:
Fixed.